external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDataToCatalyst.scala

```diff
@@ -53,8 +54,10 @@ case class AvroDataToCatalyst(
 
   @transient private lazy val avroSchema = new Schema.Parser().parse(jsonFormatSchema)
 
-  @transient private lazy val reader = new GenericDatumReader[Any](avroSchema)
-
+  @transient private lazy val reader =
+    writerJsonFormatSchema.fold(new GenericDatumReader[Any](avroSchema))(
```

Comment:
maybe a `map(...).getOrElse(..)` would be more readable.

Reply:
Changed 👍